tools/gen-intrinsics-specs.py

```diff
@@ -0,0 +1,718 @@
+    False
+    """
+    return len(row) == 6 and row[0] == __CSV_HEADER_PREFIX
+
+def is_intrinsic_definition(row):
+    return len(row) == 5
+
+
+def get_section_data(row):
+    section_text = row[2] if len(row) == 3 else None
+    return [row[1], section_text]
+
+
+def process_db(db, classification_db):
+    filtered = {}
+    section, section_text,table_header = None, None, None
+    for row in db:
+        # Set section if the line in the file is starting with __CSV_SECTION_PREFIX.
+        if is_section_command(row):
+            section, section_text = get_section_data(row)
+            if section_text:
+                recurse_set(filtered, [section],
+                            section_text, __SECTION_TEXT_KEYWORD)
+
+            print(
+                f"Activating {__CSV_SECTION_PREFIX} command at line {db.line_num}, '{section}'", file=sys.stderr)
+            continue
+
+        # Skip comment lines.
+        if is_comment_line(row):
+            continue
+
+        if is_table_header(row):
+            # Only one <HEADER> command is allowed.
+            assert(table_header is None)
+            # Set the header.
+            table_header = row[1:]
+            assert(len(table_header) == 5)
+            continue
+
+        if is_intrinsic_definition(row):
+            classification = classification_db.get(
+                get_intrinsic_name(row[0]), "No category")
+            intrinsic = Intrinsic(
+                row[0], row[1], row[2], row[3], row[4], classification)
+            classification_list = intrinsic.classification.split('|')
+            if section:
+                classification_list = [section]+classification_list
+            recurse_set(filtered, classification_list,
+                        intrinsic.table_row(), __INTRINSIC_TABLE_KEYWORD)
+
+            continue
+
+        print(f"Skipping line {db.line_num}: row = {row}", file=sys.stderr)
+    # Make sure that the CSV has at a row that have set the tale
+    # header.
+    assert(table_header is not None)
+    body = ""
+    for k, v in filtered.items():
+        body += "\n" + \
+            recurse_print_to_rst((k, v), rst_header_levels, headers=table_header, tablefmt="grid")
+    return body
+
+
+def get_classification_map(classification_file):
+    classification_map = {}
+    with open(classification_file) as csvclassificationfile:
+        classification_db = csv.reader(csvclassificationfile, delimiter='\t')
+        classification_map = {}
+        for row in classification_db:
+            if len(row) == 2:
+                classification_map[row[0]] = row[1]
+                continue
+
+            # Skip comment lines.
+            if is_comment_line(row):
+                continue
+
+            print(
+                f"{classification_file}: skipping line {classification_db.line_num}: row = {row}", file=sys.stderr)
+
+    return classification_map
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="TODO")
```

Comment:
Did you mean to fill in these TODOs, so that the `--help` output actually helps?

Perhaps a useful thing to put in the help for the three pathname options would be an example file in this repo that can be sensibly used with that option. That would be enough to let a user figure out a useful command line from scratch.

Reply:
I have filled the TODO, but instead of adding a self explanatory example with CSV files I have added a self explanatory unit tests for `process_db`. Let me know if you think this id not as useful as you intended with the original suggestion.